phylo/src/alignment/sequences.rs

```diff
@@ -35,6 +35,25 @@ impl Sequences {
         }
     }
 
+    pub fn with_alphabet(s: Vec<Record>, alphabet: Alphabet) -> Sequences {
+        let len = if s.is_empty() { 0 } else { s[0].seq().len() };
+        if s.iter().filter(|rec| rec.seq().len() != len).count() == 0 {
```

Comment:
you can optionally `.skip(1)`, as `len == s[0].seq().len()`

Reply:
Also, please consider using `Iterator::any` or (`Iteartor::all`) instead of `Iterator::count`:
```rust
if s.iter().skip(1).all(|r| r.seq().len() == len) {
  ...
} else {
  ...
}
```

Reply:
Done